infrastructure/container/container_test.go

```diff
@@ -0,0 +1,391 @@
+/*
+ * © 2025 Snyk Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package container
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/snyk/snyk-ls/application/config"
+	"github.com/snyk/snyk-ls/internal/observability/error_reporting"
+	"github.com/snyk/snyk-ls/internal/product"
+	"github.com/snyk/snyk-ls/internal/types"
+)
+
+// MockRoundTripper is a mock HTTP transport for testing
+type MockRoundTripper struct {
+	Response *http.Response
+	Error    error
+}
+
+func (m *MockRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	if m.Error != nil {
+		return nil, m.Error
+	}
+	return m.Response, nil
+}
+
+// mockHttpClientWithResponse returns a mock HTTP client that returns the given response
+func mockHttpClientWithResponse(statusCode int, body string) func() *http.Client {
+	return func() *http.Client {
+		return &http.Client{
+			Transport: &MockRoundTripper{
+				Response: &http.Response{
+					StatusCode: statusCode,
+					Body:       io.NopCloser(bytes.NewBufferString(body)),
+					Header:     make(http.Header),
+				},
+			},
+		}
+	}
+}
+
+// mockHttpClient returns a mock HTTP client with default successful response
+func mockHttpClient() func() *http.Client {
+	mockResponse := `{
+		"baseImage": "ubuntu:18.04",
+		"recommendations": [
+			{
+				"image": "ubuntu:22.04-slim",
+				"reason": "Latest LTS version with security patches"
+			}
+		],
+		"minorUpgrades": [
+			{
+				"image": "ubuntu:18.04.6",
+				"reason": "Minor version upgrade with security fixes"
+			}
+		],
+		"majorUpgrades": [
+			{
+				"image": "ubuntu:22.04",
+				"reason": "Major version upgrade"
+			}
+		],
+		"alternativeImages": []
+	}`
+	return mockHttpClientWithResponse(200, mockResponse)
+}
+
+func TestScanner_Product(t *testing.T) {
+	c := config.New()
```

Comment:
we have test helpers that help with setting up a new config, e.g. c := testutil.UnitTest(t) for a unit test or `c:= testutil.SmokeTest(t)` for a smoke test that tests E2E.